### DISCUSSION TITLE
@types/react-datepicker: popperModifiers incompatible with Popper v2 used by react-datepicker v4

### DISCUSSION THREAD

**Discussion by fdcds** [DefinitelyTyped/DefinitelyTyped#54095] [Issues with a @types package] [answered]
The types for react-datepicker appear to not reflect the switch to Popper v2 that came with react-datepicker v4. The `popperModifiers`, that were an object in react-datepicker v3 / Popper v1, are now an array: https://popper.js.org/docs/v2/migration-guide/#10-update-modifiers

See-also: https://github.com/Hacker0x01/react-datepicker/issues/2935

**Comment by tregusti** [comment]
The dependency of this TD-package also uses the old popper.js package (v1) instead of @popperjs/core (v2).

**Comment by fdcds** [comment, marked as answer]
I cannot find a changelog or release notes or even VCS tags, but I believe this is fixed in @types/react-datepicker v4.1.4 which I assume based on the commit date matching the release date contains https://github.com/DefinitelyTyped/DefinitelyTyped/commit/296d2df84bb6a161976613c5560cd0f5595ecff9
